src/pykeen/triples/instances.py

```diff
@@ -27,34 +27,22 @@
 class Instances(data.Dataset):
     """Triples and mappings to their indices."""
 
-    #: A PyTorch tensor of triples
-    mapped_triples: MappedTriples
-
-    #: A mapping from relation labels to integer identifiers
-    entity_to_id: EntityMapping
-
-    #: A mapping from relation labels to integer identifiers
-    relation_to_id: RelationMapping
-
-    @property
-    def num_instances(self) -> int:  # noqa: D401
+    def __len__(self):
         """The number of instances."""
-        return self.mapped_triples.shape[0]
-
-    @property
-    def num_entities(self) -> int:  # noqa: D401
-        """The number of entities."""
-        return len(self.entity_to_id)
-
-    def __len__(self):  # noqa: D105
-        return self.num_instances
+        raise NotImplementedError
 
 
 @fix_dataclass_init_docs
 @dataclass
 class SLCWAInstances(Instances):
     """Triples and mappings to their indices for sLCWA."""
 
+    #: The mapped triples, shape: (num_triples, 3)
+    mapped_triples: MappedTriples
+
+    def __len__(self):  # noqa: D105
+        return self.mapped_triples.shape[0]
+
```

Comment:
this one should also get a `from_triples` class method even though it will be trivial

Reply:
okay maybe this isn't important since there's an adapter pattern used to instantiate these

Reply:
Also, we already have a kind of factory, when creating them with `create_*_instance` from the TriplesFactory and/or TrainingLoop